server/test/api/depots.js

```diff
@@ -0,0 +1,108 @@
+/* jshint expr: true */
+/* global describe, it, beforeEach */
+
+var chai = require('chai');
+var expect = chai.expect;
+
+/** import test helpers */
+var helpers = require('./helpers');
+var uuid    = require('node-uuid');
+helpers.configure(chai);
+
+
+/**
+* The /depots API endpoint
+*
+* @desc This test suit is about the crud operation with depots
+*
+*/
+describe('The /depots API endpoint :: ', function () {
+  'use strict';
+
+  var agent = chai.request.agent(helpers.baseUrl);
+
+  /** login before each request */
+  beforeEach(helpers.login(agent));
+
+  /** new depot object */
+  var newDepot = {
+    uuid : uuid.v4(),
+    /** the reference column is auto increment */
+    text : 'New Depot',
+    enterprise_id : 1,
+    is_warehouse : 0
+  };
+
+  /** debot object with missing uuid */
+  var badDepot = {
+    text : 'New Depot',
+    enterprise_id : 1,
+    is_warehouse : 0
+  };
+
+  /** update depot */
+  var editDepot = {
+    uuid : uuid.v4(),
+    text : 'Edited Depot',
+    is_warehouse : 1
+  };
+
+  it('POST /depots create a new depot in the database', function () {
+    return agent.post('/depots')
+    .send(newDepot)
+    .then(function (res) {
+      expect(res).to.have.status(201);
+      expect(res.body.id).to.exist;
```

Comment:
This should check for a `uuid` instead of an `id`.

If you want to do this test automatically, simply use
```js
helpers.api.created(res);
```
See the [API testing reference](https://github.com/IMA-WorldHealth/bhima-2.X/wiki/Testing-APIs) for more details.